### PR TITLE
feat(events): RHINENG-1789 push additional headers with kafka events

### PR DESCRIPTION
**Pull request by skateman**
# Overview

This PR is being created to address [RHINENG-1789](https://issues.redhat.com/browse/RHINENG-1789).
We need some additional headers for the emitted kafka messages upon database changes for a more fine-grained filtering in cyndi. Vulnerability and patch require the `reporter` field to be exposed as a header, other apps need additinally the OS name and whether the host is an edge host or not.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by thearifismail — CHANGES_REQUESTED** [review]
@skateman this change has wider implications than these 2 files.  First, database creation failed, which if fixed by adding `from app.models import Host` created the DB but caused 179 out 1120 tests to fail. 
Since it has pretty big implications, I suggest creating a Jira card so the issue could be addressed appropriately.

**Review by kruai — CHANGES_REQUESTED** [review]
There are places where a string is still being sent to `message_headers` rather than the full host. The unit test suite show 178 failures and 30 errors. If you're having trouble running the tests locally, let us know and we can help you out

**Comment by skateman** [comment]
Yup, will work on that

**Comment by skateman** [comment]
@kruai @thearifismail I have no idea why the remaining tests are failing and don't even thave the opportunity do debug it as the developer setup documentation you have in the README doesn't work for me (neither does the docker build). Can you please either help me out with a developer container or point me in the right direction why are tests failing? 🙏

File: api/host.py [hunk 1]
```diff
@@ -299,10 +299,16 @@ def get_host_system_profile_by_id(
     return flask_json_response(json_output)
 
 
-def _emit_patch_event(serialized_host, host_id, insights_id):
-    headers = message_headers(EventType.updated, insights_id)
+def _emit_patch_event(serialized_host, host):
+    headers = message_headers(
+        EventType.updated,
+        host.canonical_facts.get("insights_id"),
+        host.reporter,
```
Comment (vkrizan) [inline api/host.py:306]:
```
Traceback (most recent call last):
  File "/home/vkrizan/insights/insights-host-inventory/rebuild_events_topic.py", line 64, in run
    sync_event_message(json.loads(message.value()), session, event_producer)
  File "/home/vkrizan/insights/insights-host-inventory/app/queue/queue.py", line 204, in sync_event_message
    host.reporter,
AttributeError: 'LimitedHost' object has no attribute 'reporter'
```

**Comment by thearifismail** [comment]
> Yup, will work on that

Looks like you got it working because I just checked out the `adjust-headers` branch and it passed all tests.

**Comment by thearifismail** [comment]
I asked @fstavela to look into it because IQE tests are failing

**Review by kruai — APPROVED** [review]
The remaining IQE test failures appear to be due to intermittent network issues (remote disconnect stuff). I'd like to get another approval before merging, but this looks fine to me.

**Review by thearifismail — APPROVED** [review]

**Review by fstavela — APPROVED** [review]
LGTM

**Pull request merged**